src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,37 @@
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
+
+import { getUser } from '../../api';
+
+import { TodoContext } from '../../context/ContextTodo';
+
+import { User } from '../../types/User';
+
 import { Loader } from '../Loader';
 
 export const TodoModal: React.FC = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  const {
+    setIsModalOpen,
+    selectedTodo,
+    setSelectedTodo,
+  } = useContext(TodoContext);
+
+  useEffect(() => {
+    getUser(selectedTodo?.userId ?? 0)
+      .then(user => setSelectedUser(user));
+  }, []);
+
+  const handleCloseModal = () => {
+    setSelectedTodo(null);
+    setIsModalOpen(false);
+  };
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {!selectedUser ? (
```

Comment:
Better to add `isLoading` state as in `App` component

Reply:
Not fixed

Reply:
If you could, please explain why it's better. For me it's understandable and it works

Reply:
@FrontKid You should try to structure the code in such a way that the actions in it look logical.
When you do a loading, first of all you expect that it will complete, not that the user will choose.